docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+This list of requirements is not exhaustive, but instead focuses on the requirements which will drive our decision making process.
+As such, it identifies the non-requirements first.
+
+### Not a requirement
+The following items all may have some degree of desirability, but they are not hard requirements for the initial version
+* While the https connection to the server must be validated, there is no requirement to verify the content received by the server - ie, there’s no requirement to check the signature of the body itself.
+* There’s no requirement to validate the result of the server conforms to a pre-defined schema - we trust the server data.
+* There’s no requirement to return strongly-typed data to the applications - returning a JSON string/object is suitable.
+* There’s no requirement to cache server responses to the file-system - if the app requests content, it’s fine for the library to always hit the server.
+* There’s no requirement for any kind of scheduling or awareness of network state - when we are requested for content, we do it immediately and return an appropriate error if it can not be fetched.
+* There’s no requirement to support publishing records, requesting reviews or providing approvals via this new library.
+* There’s no requirement to manage buckets, groups and collections via this new library.
+
+### Requirements
+The requirements we do have are:
+* The library should allow fetching records from Mozilla’s Remote Settings servers. This includes support for attachments, and fetching [incremental changes](https://docs.kinto-storage.org/en/latest/api/1.x/filtering.html#polling-for-changes).
+* The library should not create threads or run any event loops - the mobile apps themselves are responsible for all threading requirements. While this might change in the future, considering this kind of change to our mobile applications is out of scope for this project.
+* We must use Necko for all networking on Android, must enforce all connections are via valid https hosts (although some test-only exceptions might be helpful for QA, such as allowing localhost connections to be http)
+* The library should be the only remote-settings library used in the browser. Specifically, this means that Nimbus must also be capable of using the library, and the work to move Nimbus to the library must be considered as part of the project.
+
+## Existing Libraries
+We have identified the following libraries which may be suitable for this project.
+### Remote-settings on desktop
+There is a version of the [remote settings client in desktop](https://searchfox.org/mozilla-central/source/services/settings/RemoteSettingsClient.jsm), written in Javascript.
+It is obviously very capable, but the roadblock to it being suitable for use by our mobile browsers is that it is written in Javascript, so there’s no reasonable path to have it made available to Swift and Kotlin.
```

Comment:
Can we describe or link to why it's "obviously" very capable 😅 Even something along the lines of "it's been stable in desktop for X long thus it's capable"

Reply:
Hmmm there's maybe a non-option here of "Move the remote settings client on desktop to xul and expose it to android using gecko-view" that doesn't solve the iOS problem and is a lot more work and high risk.. but maybe noting it down for completion?

Reply:
Hopefully addressed both of these in3c356141e9177b963fbaf31aa07e9196b343cc27